### PR TITLE
Added Message Reactions Feature

### LINKED ISSUE #67: [Feature Request]:  Add Message Reactions to Chat
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Feature Description

I would like to propose adding a message reactions feature to the chat app. This would allow users to react to individual messages using emojis

**Benefits:**

-Enhances user engagement.
-Provides a quick way to express feedback.

Please mention this under gssoc and hacktoberfest

### Record

- [X] I agree to follow this project's Code of Conduct
- [X] I want to work on this issue
- [X] I'm willing to provide further clarification or assistance if needed.

### PR DESCRIPTION
**Pull request by muskan171105**
This PR introduces a new feature that allows users to react to messages with emojis (e.g., ❤️, 👍). Reactions are broadcasted in real-time using Socket.io, so all users in the chat can see the reactions immediately.

Changes:

Added reactMessage socket event to handle message reactions.
Updated the server to broadcast reactions to connected users.
Modified frontend to send and display reactions.

Testing:

Tested reactions in real-time chat rooms."

Closes #67

### CODE REVIEW

File: backend/server.js [hunk 1]
```diff
@@ -37,12 +36,12 @@ const io = require("socket.io")(server, {
   pingTimeout: 60000,
   cors: {
     origin: ["http://localhost:3000", "https://chat-app-two-theta-52.vercel.app", "https://chatapi.gitleet.live"],
-    // credentials: true,
```
Comment (kartikayasijaa) [inline backend/server.js:40]: Why you removed this?

Comment (muskan171105) [reply backend/server.js:40]: Wait I'll fix it

Comment (muskan171105) [reply backend/server.js:40]: It was anyways commented. So I removed it. But to handle a few functions, I will uncomment it and push it again

File: backend/server.js [hunk 2]
```diff
@@ -37,12 +36,13 @@ const io = require("socket.io")(server, {
...
+    credentials: true,  // Allow credentials
```
Comment (muskan171105) [inline backend/server.js:39]: Check this

**Review by muskan171105 — COMMENTED** [review]
Changes made

File: backend/server.js (hunk 2 above)
Comment (kartikayasijaa) [reply backend/server.js:39]: just leave it commented

Comment (muskan171105) [reply backend/server.js]: Done

**Comment by kartikayasijaa** [comment]
Are you storing these reactions in DB?

**Comment by muskan171105** [comment]
No I haven't stored them in a db. The reactions will be broadcast to other users in real time through socket.io

**Pull request merged**